Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -487,6 +487,14 @@ void GraphicsContextSkia::setupStrokeSource(SkPaint& paint) const
         paint.setColor(SkColor(strokeBrush().color().colorWithAlphaMultipliedBy(alpha())));
 }
 
+void GraphicsContextSkia::drawCustom(std::function<void(SkCanvas&)> drawFunc) const
+{
+    if (!makeGLContextCurrentIfNeeded())
+        return;
```

Comment:
This might not be enough because of the optimization that checks the rendering purpose, because in the case of filters the purpose is Unspecified. I think we should check when the Skia filters are applied. If they always happen during layer rendering, then we don't need the drawCustom, because skia GL context is already current.

Reply:
As discussed, I removed this new function from GraphicsContextSkia as it's not needed.